src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const listOfStateVersions = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperty(stateClone, action.keysToRemove);
+        break;
+
+      case 'clear':
+        clearObject(stateClone);
+        break;
+
+      default:
+        return stateClone;
```

Comment:
Instead of returning `stateClone` in the default case, it would be better to send an error if action type doesn't exist.